### ISSUE TITLE
[Rule Request] Databricks Implementation 2.0 is not allowed

### ISSUE THREAD

**Issue by ConstancaElias** [TabularEditor/BestPracticeRules#68]
I have a table in my semantic model that is connected to Azure Databricks. The connection to Databricks is defined in Power Query using the following M expression:

```
let
    Source = Databricks.Query(ServerHostname, HTTPPath, [Catalog=null, Database=Database, EnableExperimentalFlagsV1_1_0=null,Implementation="2.0"])
in
    Source
```
In Power BI Desktop, the parameter Implementation = "2.0" is required for the refresh to work correctly.
However, when the model is published to Power BI Service, the query must not include this parameter.

For this reason, I would like to create a BPA rule that validates whether a Databricks query contains the parameter Implementation = "2.0".

The expected behavior of the rule is:
- If the Databricks query contains Implementation = "2.0", the rule should raise an error.
- If the parameter is not present, the rule should pass.

I am looking for guidance on how to implement such a rule. Thank you!

**Comment by otykier** [comment]
Apologies for the late reply.

I'm thinking a simple solution could be to check that there are no M expressions in the model that contain the string `Databricks.Query` or `Databricks.Catalogs`, while simultaneously NOT containing the string `Implementation="2.0"`.

So in other words:

1. Set the "Applies to" to **Partitions** and **Named Expressions**
2. Set the rule expression to:

```linq
(Expression.Contains("Databricks.Query") or Expression.Contains("Databricks.Catalogs"))
and not Expression.Contains("Implementation=""2.0""")
```

<img width="1180" height="602" alt="Image" src="https://github.com/user-attachments/assets/60b9a36c-862d-422f-81eb-6244eabfea41" />

**Comment by otykier** [comment]
oh, sorry - I just noticed that you requested a rule that flags partitions/shared expressions that **DO** specify `Implementation=2.0`. Luckily, the solution is simple - just remove the `not` in line 2 of the expression:

```linq
(Expression.Contains("Databricks.Query") or Expression.Contains("Databricks.Catalogs"))
and Expression.Contains("Implementation=""2.0""")
```
